### PR TITLE
Add make audit

### PR DESCRIPTION
**Pull request by Hoverbear**
Add an audit command as part of #4264 .

Test by running `make audit`, regardless of a `cargo-audit` version being installed. It will install (or update to) the latest, then run audit.

A follow up PR will add this to our CI builds in the SRE repo.

## Test locally

```rust
make audit
```

You can test with or without cargo-audit installed.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -117,6 +117,19 @@ format: pre-format
 	@cargo fmt --all -- --check >/dev/null || \
 	cargo fmt --all
 
+pre-audit:
+	@cargo install cargo-audit --force
+
+LATEST_AUDIT_VERSION = $(strip $(shell cargo search cargo-audit | head -n 1 | awk '{ gsub(/"/, "", $$3); print $$3 }'))
+CURRENT_AUDIT_VERSION = $(strip $(shell (cargo audit --version 2> /dev/null || echo "noop 0") | awk '{ print $$2 }'))
+audit:
+	echo "$(strip $(LATEST_AUDIT_VERSION))"
```
Comment (zhangjinpeng1987) [inline Makefile]: The output is not pretty, it look likes:
```
jinpengzhang$ make audit
echo "0.6.1"
0.6.1
echo "0"
0
```

Comment (Hoverbear) [reply Makefile]: Whoops. Thought I cleaned that! 😆

File: Makefile [hunk 2]
```diff
@@ -117,6 +117,19 @@ format: pre-format
...
+	echo "$(strip $(CURRENT_AUDIT_VERSION))"
+ifneq ($(LATEST_AUDIT_VERSION),$(CURRENT_AUDIT_VERSION))
+	cargo install cargo-audit --force
```
Comment (breeswish) [inline Makefile]: Looks like these should be put in "pre-audit" instead?

Comment (Hoverbear) [reply Makefile]: You're right!

File: Makefile [hunk 3]
```diff
@@ -117,8 +117,18 @@ format: pre-format
 	@cargo fmt --all -- --check >/dev/null || \
 	cargo fmt --all
 
+LATEST_AUDIT_VERSION = $(strip $(shell cargo search cargo-audit | head -n 1 | awk '{ gsub(/"/, "", $$3); print $$3 }'))
```
Comment (Connor1996) [inline Makefile]: seems other variables all at the beginning of this file, maybe we should them consistent

**Comment by Hoverbear** [comment]
\retry

**Comment by Hoverbear** [comment]
This failed in our CI because openssl headers weren't installed: https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tikv_ghpr_test/detail/tikv_ghpr_test/2030/pipeline/

:(

**Pull request merged**